### DISCUSSION TITLE
Hierarchical navigation for the documentation pages

### DISCUSSION THREAD

**Discussion by v479038280** [zauberzeug/nicegui#2976] [Ideas / Feature Requests]
The current navigation page is not very convenient or intuitive. You have to open each subpage to see the specific elements it contains, and you also have to open each element to find out if there is a 'more' section and what the 'more' sections are. I think the following type of navigation would be more convenient, as it allows you to directly see all the items and their sub-content. This way, you can immediately identify the items that are of interest, and choose not to look at those that are not.

![微信截图_20240429160134](https://github.com/zauberzeug/nicegui/assets/25096045/ce625c59-ec86-4b65-b929-453c2ddcfcb9)
![微信截图_20240429160034](https://github.com/zauberzeug/nicegui/assets/25096045/b34099de-5f84-48b0-9361-eed0532fdae5)

**Comment by falkoschindler** [comment]
I agree. Having direct access to the whole hierarchy of documentation pages in a sidebar would be great.
We'll have to think about how this could be added to the existing infrastructure.

**Reply by evnchn** [reply]
Got it done in #4732
